File: algorithms/php/744.FindSmallestLetterGreaterThanTarget.php
<?php

class Solution
{

    /**
     * @param String[] $letters
     * @param String $target
     * @return String
     */
    function nextGreatestLetter($letters, $target)
    {
        $l = 0;
        $r = count($letters) - 1;
        $target = ord($target);

        while ($l <= $r) {
            $m = intdiv($l + $r, 2);

            if (ord($letters[$m]) <= $target) {
                $l = $m + 1;
            } else {
                $r = $m - 1;
            }
        }

        if ($l > count($letters) - 1) {
            $l = 0;
        }
        return $letters[$l];
    }
}

File: algorithms/cpp/744.FindSmallestLetterGreaterThanTarget.cpp
class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        int l = 0;
        int r = letters.size() - 1;

        while (l <= r) {
            int m = l + (r - l) / 2;

            if (letters[m] <= target) {
                l = m + 1;
            } else {
                r = m - 1;
            }
        }

        if (l > letters.size() - 1) {
            l = 0;
        }

        return letters[l];
    }
};